File: src/main/java/org/mitallast/queue/security/ECDHEncrypted.java
package org.mitallast.queue.security;

import org.mitallast.queue.common.codec.Codec;
import org.mitallast.queue.common.codec.Message;

public class ECDHEncrypted implements Message {
    public static final Codec<ECDHEncrypted> codec = Codec.Companion.of(
            ECDHEncrypted::new,
            ECDHEncrypted::sign,
            ECDHEncrypted::iv,
            ECDHEncrypted::encrypted,
            Codec.Companion.bytesCodec(),
            Codec.Companion.bytesCodec(),
            Codec.Companion.bytesCodec()
    );

    private final byte[] sign;
    private final byte[] iv;
    private final byte[] encrypted;

    public ECDHEncrypted(byte[] sign, byte[] iv, byte[] encrypted) {
        this.sign = sign;
        this.iv = iv;
        this.encrypted = encrypted;
    }

    public byte[] sign() {
        return sign;
    }

    public byte[] iv() {
        return iv;
    }

    public byte[] encrypted() {
        return encrypted;
    }
}

File: src/main/java/org/mitallast/queue/security/ECDHEncrypted.kt
package org.mitallast.queue.security

import io.netty.buffer.ByteBuf
import org.mitallast.queue.common.codec.Codec
import org.mitallast.queue.common.codec.Message

class ECDHEncrypted(
    val sign: ByteArray,
    val iv: ByteArray,
    val len: Int,
    val encrypted: ByteBuf) : Message {

    companion object {
        val codec = Codec.of(
            ::ECDHEncrypted,
            ECDHEncrypted::sign,
            ECDHEncrypted::iv,
            ECDHEncrypted::len,
            ECDHEncrypted::encrypted,
            Codec.bytesCodec(),
            Codec.bytesCodec(),
            Codec.intCodec(),
            Codec.byteBufCodec()
        )
    }
}
